src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -399,6 +414,166 @@ void ZeroInferRequest::set_tensor(const ov::Output<const ov::Node>& port, const
+                        " and shape ",
+                        item_shape,
+                        " is not compatible with batched tensor with element type ",
+                        element_type,
+                        " and shape ",
+                        batched_shape);
+        OPENVINO_ASSERT(item->is_continuous(), "Strides for batched tensors should be default.");
+    }
+}
+
+void ZeroInferRequest::set_tensors(const ov::Output<const ov::Node>& port,
+                                   const std::vector<ov::SoPtr<ov::ITensor>>& tensors) {
+    OV_ITT_SCOPED_TASK(itt::domains::LevelZeroBackend, "set_tensors");
+    if (tensors.size() == 1) {
+        set_tensor(port, tensors[0]);
+        return;
+    }
+
+    auto foundPort = find_port(port);
+    OPENVINO_ASSERT(foundPort.found(), "Cannot find input tensor for port ", port);
+    if (!foundPort.is_input()) {
+        OPENVINO_THROW("set_input_tensors/set_tensors is not supported for output port.");
+    }
+
+    check_batched_tensors(port, tensors);
+
+    _userBatchedTensors[foundPort.idx] = tensors;
+
+    if (_initStructs->getMutableCommandListVersion()) {
+        if (_batchSize.has_value()) {
+            for (size_t i = 0; i < tensors.size(); i++) {
+                auto remoteTensor = std::dynamic_pointer_cast<ZeroRemoteTensor>(tensors[i]._ptr);
+
+                _levelZeroBatchedTensors[foundPort.idx].resize(tensors.size());
+                _batchedTensorsData[foundPort.idx].resize(tensors.size());
+
+                if (remoteTensor == nullptr) {
+                    bool tensorL0Context = false;
+
+                    ze_memory_allocation_properties_t desc = {};
+                    desc.stype = ZE_STRUCTURE_TYPE_MEMORY_ALLOCATION_PROPERTIES;
+                    auto res = zeMemGetAllocProperties(_initStructs->getContext(), tensors[i]->data(), &desc, nullptr);
+                    if (res == ZE_RESULT_SUCCESS) {
+                        if (desc.id) {
+                            switch (desc.type) {
+                            case ZE_MEMORY_TYPE_HOST:
+                            case ZE_MEMORY_TYPE_DEVICE:
+                            case ZE_MEMORY_TYPE_SHARED:
+                                _logger.debug(
+                                    "ZeroInferRequest::set_tensors - tensor was created in the same L0 context");
+
+                                _levelZeroBatchedTensors[foundPort.idx][i] = tensors[i];
+                                tensorL0Context = true;
+                                break;
+                            case ZE_MEMORY_TYPE_UNKNOWN:
+                            case ZE_MEMORY_TYPE_FORCE_UINT32:
+                            default:
+                                break;
+                            }
+                        }
+                    }
+
+                    if (!tensorL0Context) {
+                        _logger.debug("ZeroInferRequest::set_tensors - tensor wasn't created in the same L0 context, "
```

Comment:
Can you consider adding a warning message that it implies data copying?

Reply:
not sure that we want a warning here. isn't a warning or a problem.